src/modules/hide_sidebar_elements/index.js

```diff
@@ -28,6 +52,29 @@ class HideSidebarElementsModule {
         $('body').toggleClass('bttv-hide-featured-channels', settings.get('hideFeaturedChannels'));
     }
 
+    toggleAutoExpandChannels() {
+        if (settings.get('autoExpandChannels') === true) {
+            $('.side-nav-load-more__button').trigger('click');
+        }
+    }
+
+    toggleRecommendedFriends() {
+        $('body').toggleClass('bttv-hide-recommended-friends', settings.get('hideRecommendedFriends'));
+    }
+
+    toggleOfflineFollowedChannels() {
+        var resizeTimer = null;
+        function hideOfflineUser() {
+            if (settings.get('hideOfflineFollowedChannels') === true) {
+                $('.side-nav-card__avatar--offline').parent().css('cssText', 'display: none!important');
+            }
+        };
+        $(window).bind('resize', function() {
```

Comment:
not a fan of this solution. is there not a way to do this using css alone? if so, can you explain?

Reply:
in the old layout, yes. but from what I can gather, the best way to determine offline status thru css is 2 layers too deep meaning some jquery can do the trick at finding the desired element. 

This would be the end of it if resizing the window didn't cause every single element to delete and redraw itself clearing out the old css overrides.

From what I've found, the best solution is to wait for [CSS4 Selector Overrides](https://caniuse.com/#feat=css-has) to be supported by literally any single browser and not have to rely on Javascript to do this unless a better Javascript solution exists that I don't know of yet

Reply:
couldn't you just apply a css class that performs a display none?

Reply:
no since if you apply it to the class which has the flag for offline, it'll only hide the profile picture. you'd need the parent of the class with the offline flag and there isn't a supported parent selector in css (yet)

[this is the layout for offline user](https://i.imgur.com/rfTIn4X.png) while [this is the layout for an online user](https://i.imgur.com/um5jloc.png).

The only discernable difference is the --offline flag in the avatar element :/

Reply:
I meant use js to apply the class.. it shouldn't cause you issues on resize then